sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala

```diff
@@ -1719,7 +1719,7 @@ class ChunkedPacker(table: Table,
   }
 
   override def next(): (DeviceBounceBuffer, Long) = {
-    withResource(bounceBufferPool.nextBuffer()) { bounceBuffer =>
+    closeOnExcept(bounceBufferPool.nextBuffer()) { bounceBuffer =>
```

Comment:
So just to confirm that the bounce buffer will be closed by the caller of next? I am just confused because that should have caused a double free of some kind because we closed it and then returned it.

Reply:
yeah this is `DeviceBounceBuffer`, it is not `DeviceMemoryBuffer`. I can add a throw in `DeviceBounceBuffer` if we unacquire too many times causing the double free. The bug is that it returns the buffer back to the pool and now it is eligible for the next write.. doing bad bad things.

Reply:
That is fine, it is just confusing that we only "fixed" one side of it.